### PR TITLE
Remove command-line options and configuration fields

### PR DESCRIPTION
**Pull request by stevenengler**
Removes the following:

Command-line options:
- `--preload`
- `--valgrind`
- `--test`
- `--tgen`
- `--cpu-precision`
- `--cpu-threshold`
- `--tcp-ssthresh`
- `--tcp-windows`

Configuration fields:
- `cpufrequency`

Part of #1220.

### CODE REVIEW

File: src/main/core/controller.c [hunk 1]
```diff
@@ -348,20 +348,16 @@ static void _controller_registerHostCallback(ConfigurationHostElement* he, Contr
...
         if (params->cpuFrequency == 0) {
             params->cpuFrequency = 2500000; // 2.5 GHz
             debug("both configured and raw manager cpu frequencies unavailable, using 2500000 KHz");
         }
 
-        gint defaultCPUThreshold = options_getCPUThreshold(controller->options);
-        params->cpuThreshold = defaultCPUThreshold > 0 ? defaultCPUThreshold : 0;
-        gint defaultCPUPrecision = options_getCPUPrecision(controller->options);
-        params->cpuPrecision = defaultCPUPrecision > 0 ? defaultCPUPrecision : 0;
+        params->cpuThreshold = 0;
```
Comment (robgjansen) [inline src/main/core/controller.c:353]: Should this be 0 or -1? It looks like -1 was the default before to disable cpu delays?

File: src/main/core/support/options.c [hunk 2]
```diff
@@ -172,8 +169,6 @@ Options* options_new(gint argc, gchar* argv[]) {
       { "socket-recv-buffer", 0, 0, G_OPTION_ARG_INT, &(options->initialSocketReceiveBufferSize), sockrecv->str, "N" },
       { "socket-send-buffer", 0, 0, G_OPTION_ARG_INT, &(options->initialSocketSendBufferSize), socksend->str, "N" },
       { "tcp-congestion-control", 0, 0, G_OPTION_ARG_STRING, &(options->tcpCongestionControl), "Congestion control algorithm to use for TCP ('aimd', 'reno', 'cubic') ['reno']", "TCPCC" },
-      { "tcp-ssthresh", 0, 0, G_OPTION_ARG_INT, &(options->tcpSlowStartThreshold), "Set TCP ssthresh value instead of discovering it via packet loss or hystart [0]", "N" },
-      { "tcp-windows", 0, 0, G_OPTION_ARG_INT, &(options->initialTCPWindow), "Initialize the TCP send, receive, and congestion windows to N packets [10]", "N" },
```
Comment (robgjansen) [inline src/main/core/support/options.c:182]: The [getting started tutorial doc](https://github.com/shadow/shadow/blob/dev/docs/2-Getting-Started-Tutorial.md#example-experiment) has an example that tweaks the `--tcp-windows` option. We should update that example experiment so it no longer references a non-existent option.

File: src/main/core/support/options.c [hunk 3]
```diff
@@ -385,11 +378,6 @@ gboolean options_shouldExitAfterShmCleanup(Options* options) {
     return options->shouldExitAfterShmCleanup;
 }
 
-const gchar* options_getPreloadString(Options* options) {
```
Comment (robgjansen) [inline src/main/core/support/options.c:438]: Huh, this wasn't called anywhere?

**Review by robgjansen — APPROVED** [review]
Always happy to see a bunch of old code removed :)

File: src/main/core/controller.c (hunk 1 above)
Comment (stevenengler) [reply src/main/core/controller.c:353]: The `params->cpuThreshold = defaultCPUThreshold > 0 ? defaultCPUThreshold : 0` line sets a lower limit of 0, so the -1 would end up as 0 here.

File: src/main/core/support/options.c (hunk 3 above)
Comment (stevenengler) [reply src/main/core/support/options.c:438]: Nope, it looks like all uses were removed in the initial e41c4a997.

File: src/main/core/support/options.c (hunk 2 above)
Comment (stevenengler) [reply src/main/core/support/options.c:182]: I updated this example to use different queuing disciplines instead. Let me know if you think there's a better option to use.

**Pull request merged**